### PR TITLE
[ONNX] Convert aten numpy_T to ONNX transpose

### LINKED ISSUE #51183: RuntimeError: Exporting the operator numpy_T to ONNX opset version 9, 10, 11 & 12 is not supported
## 🐛 Bug

RuntimeError: Exporting the operator numpy_T to ONNX opset version 9, 10, 11 & 12 is not supported. Please open a bug to request ONNX export support for the missing operator.

## To Reproduce

i have a error message

Traceback (most recent call last):
  File "model2onnx.py", line 69, in <module>
    main()
  File "model2onnx.py", line 63, in main
    onnx_export(model, args.output_path, input)
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 26, in decorate_context
    return func(*args, **kwargs)
  File "model2onnx.py", line 51, in onnx_export
    torch_onnx.export(model, dummy_input, output_dir,
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/__init__.py", line 225, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 85, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 632, in _export
    _model_to_graph(model, args, verbose, input_names,
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 417, in _model_to_graph
    graph = _optimize_graph(graph, operator_export_type,
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 203, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/__init__.py", line 263, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 930, in _run_symbolic_function
    symbolic_fn = _find_symbolic_in_registry(domain, op_name, opset_version, operator_export_type)
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/utils.py", line 888, in _find_symbolic_in_registry
    return sym_registry.get_registered_op(op_name, domain, opset_version)
  File "/home/ramos/miniconda3/envs/torch/lib/python3.8/site-packages/torch/onnx/symbolic_registry.py", line 111, in get_registered_op
    raise RuntimeError(msg)
RuntimeError: Exporting the operator numpy_T to ONNX opset version 12 is not supported. Please open a bug to request ONNX export support for the missing operator.

## Expected behavior

torch_onnx.export(model, dummy_input, output_dir,
                      input_names=input_names, output_names=output_names,opset_version=12)

## Environment

PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.18.2

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 9.1.85
GPU models and configuration:
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce RTX 2080 Ti
GPU 2: GeForce RTX 2080 Ti
GPU 3: GeForce RTX 2080 Ti
GPU 4: GeForce RTX 2080 Ti
GPU 5: GeForce RTX 2080 Ti
GPU 6: GeForce RTX 2080 Ti
GPU 7: GeForce RTX 2080 Ti

Nvidia driver version: 450.102.04
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.4
[pip3] torch==1.7.0
[pip3] torch2trt==0.1.0
[pip3] torchgeometry==0.1.2
[pip3] torchvision==0.8.1
[conda] cudatoolkit               10.0.130                      0
[conda] numpy                     1.19.4                   pypi_0    pypi
[conda] torch                     1.7.0                    pypi_0    pypi
[conda] torch2trt                 0.1.0                    pypi_0    pypi
[conda] torchgeometry             0.1.2                    pypi_0    pypi
[conda] torchvision               0.8.1                    pypi_0    pypi

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

### PR DESCRIPTION
**Pull request by siboehm**
This PR addes support for exporting `aten::numpy_T` to ONNX

Fixes #51183

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Thank you for the fix! We will merge it after independent test failures are fixed

**Comment by justinchuby** [comment]
@siboehm could you rebase from master?

**Review by justinchuby — CHANGES_REQUESTED** [review]
Could you also help add a test in https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_onnxruntime.py? It can be similar to 

https://github.com/pytorch/pytorch/blob/cb8222493031182c22dce2c06ce23d6e4613eba4/test/onnx/test_pytorch_onnx_onnxruntime.py#L1388-L1394

Thank you!

**Comment by siboehm** [comment]
@justinchuby CI is green and I added the test you requested, should be good for merge :)

**Review by justinchuby — APPROVED** [review]
Thanks a lot! Could you test scalar inputs as well? For example `torch.tensor(42)`.

File: test/onnx/test_pytorch_onnx_onnxruntime.py
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1305]: I would test a scalar e.g. `torch.tensor(42)` as well

Comment (siboehm) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:1305]: Done. I actually removed the 1D test case, since as per [1.11 Release Notes](https://github.com/pytorch/pytorch/releases) using `.T` on anything that isn't 0D or 2D is deprecated.

**Comment by siboehm** [comment]
The single CI failure is unrelated (an issue with setting up the Docker image)

**Review by justinchuby — APPROVED** [review]
LGTM🚀

File: test/onnx/test_operators.py
Comment (garymm) [inline test/onnx/test_operators.py:209]: Please don't add this. The expect files are a maintenance burden and we shouldn't add any more, especially if we have another way to test a feature (as we do in this case).

File: test/onnx/test_pytorch_onnx_onnxruntime.py
Comment (garymm) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1303]: No need for temporary variables
```suggestion
        self.run_test(NumpyTranspose(), torch.randn(4, 7))
```

**Review by garymm — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py
Comment (justinchuby) [reply torch/onnx/symbolic_opset9.py:96]: Sure!

**Review by garymm — APPROVED** [review]
Approved but don't merge until all the ONNX checks pass.

**Comment by siboehm** [comment]
Rebased on master in the hopes of resolving the Docker image issue that's causing CI to fail.

**Pull request closed without merging**
